core/src/main/scala/kafka/utils/Throttler.scala

```diff
@@ -83,6 +86,17 @@ class Throttler(@volatile var desiredRatePerSec: Double,
   def updateDesiredRatePerSec(updatedDesiredRatePerSec: Double): Unit = {
     desiredRatePerSec = updatedDesiredRatePerSec;
   }
+
+  // initial start time if it haven't been set. This method should be called when starting do disk IO works.
+  def initStartTimeNs(): Unit = {
+    if (periodStartNs == 0) {
+      lock synchronized {
+        if (periodStartNs == 0) {
+          periodStartNs = time.nanoseconds()
+        }
+      }
+    }
+  }
 }
 
 object Throttler {
```

Comment:
I'd like to remove all this `object Throttler {` block. Not sure if it is safe?

Reply:
Why do you want to remove it, there is a `main` method in it.

Reply:
Or are you saying that there's no benefit in there being a `main` for this? If so, then we could indeed remove it. We must make sure there is no script relying on it though.